spec.html

```diff
@@ -25894,30 +26039,44 @@ <h1>Static Semantics: ModuleRequests ( ): a List of Strings</h1>
         </emu-alg>
         <emu-grammar>ModuleItemList : ModuleItemList ModuleItem</emu-grammar>
         <emu-alg>
-          1. Let _moduleNames_ be ModuleRequests of |ModuleItemList|.
-          1. Let _additionalNames_ be ModuleRequests of |ModuleItem|.
-          1. For each String _name_ of _additionalNames_, do
-            1. If _moduleNames_ does not contain _name_, then
-              1. Append _name_ to _moduleNames_.
-          1. Return _moduleNames_.
+          1. Let _requests_ be ModuleRequests of |ModuleItemList|.
+          1. Let _additionalRequests_ be ModuleRequests of |ModuleItem|.
+          1. For each ModuleRequest Record _mr_ of _additionalRequests_, do
+            1. If _requests_ does not contain a ModuleRequest Record _mr2_ such that _mr_.[[Specifier]] is _mr2_.[[Specifier]] and ImportAttributesEqual(_mr_.[[Attributes]], _mr2_.[[Attributes]]) is *true*, then
```

Comment:
Syntactically, this form is ambiguous:
- `If _requests_ does not contain a <foo> such that (<A> and <B>)`
vs
- `If (_requests_ does not contain a <foo> such that <A>) and <B>`

although the 'scoping' of `_mr2_` only admits the first.

One possibility would be to say `... such that both <A> and <B>`, but I think the result sounds a bit odd.

Reply:
I'm solving this by replacing `ImportAttributesEqual` with a `ModuleRequestEqual` AO.